apps/test/netsim/NetSimLocalClientNode.js

```diff
@@ -38,6 +38,40 @@ describe("NetSimLocalClientNode", function () {
     assert(undefined !== testRemoteNode, "Made a remote node");
   });
 
+  describe ("onNodeTableChange_", function () {
+    beforeEach(function () {
+      testLocalNode.initializeSimulation(null, null);
+    });
+
+    it ("detects when own row has gone away and calls lost connection callback", function () {
+      var lostConnection = false;
+      testLocalNode.setLostConnectionCallback(function () {
+        lostConnection = true;
+      });
+      assertEqual(false, lostConnection);
+
+      testShard.nodeTable.api_.remoteTable.deleteMany([testLocalNode.entityID], function () {});
+      testShard.nodeTable.refresh();
+      assertEqual(true, lostConnection);
+    });
+
+    it ("also detects if row uuid changes", function () {
+      var lostConnection = false;
+      testLocalNode.setLostConnectionCallback(function () {
+        lostConnection = true;
+      });
+      assertEqual(false, lostConnection);
+
+      // Reset fake remote table and repopulate first two rows.
```

Comment:
It would be nice if the "also detects if row uuid changes" test mirrored the previous test, and the "detects when shard has been reset even if other nodes claimed own id" test was a third test